### PR TITLE
Get v1 data - the magic get functions for od

### PR DESCRIPTION
**Pull request by e-kotov**
Some further updates on v1 data.

Completed first function to smartly load and connect to a folder of cached origin-destination csv files.

See usage below + the draft vignette in the vignettes folder + help for `spod_duckdb_od_v1()` (not exported, see `duckdb_helpers.R` file) and `spod_get_od_v1()`.

`spod_get_od_v1()` is supposed to work like magic auto downloading files on the fly with underlying `spod_download_data()` and even more magically quickly connecting to them using `DuckDB` with on-the-fly translation to English of both column names and column values, as well as filtering by the specified dates no matter how many files have been cached.

```r
# Example usage
# devtools::load_all()
# remotes::install_github("Robinlovelace/spanishoddata@791da4a43a9dc747083f47591e2cef4bfba362a2", force = TRUE)

Sys.setenv(SPANISH_OD_DATA_DIR = "path/to/your/datafolder")
dates <- c("2020-02-14", "2020-03-14", "2021-02-14", "2021-02-14", "2021-02-15")
od_dist <- spod_get_od_v1(zones = "distr", dates = dates)


# this od object is a lazy table with conneciton to duckdb, so it is not loaded into memory
od_dist |> head()
od_dist |> dplyr::glimpse()


od_dist_tbl <- od_dist |> dplyr::collect() # load into memory, warning it is 1.5 GB
format(object.size(od_tbl), units = "Mb")

# to disconnect from the database properly
# but this is not necessary as the connection is readonly
# and is automatically closed and garbage collected in certain circumstances
DBI::dbDisconnect(od_dist$src$con)
rm(od_dist)


# with municipalities
od_muni <- spod_get_od_v1(zones = "muni", dates = dates)
od_muni |> glimpse()
DBI::dbDisconnect(od_muni$src$con)
rm(od_muni)
```

Also enabled the quarto rendering of vignettes.

### CODE REVIEW

**Comment by e-kotov** [comment]
Added a convert function. It uses all cached data and creates a clean duckdb file form it. Uses at least 3GB RAM, fails on 2GB RAM.

```
devtools::load_all()
Sys.setenv(SPANISH_OD_DATA_DIR = "path/to/your/datafolder")
dates <- c("2020-02-14", "2020-03-14", "2021-02-14", "2021-02-14", "2021-02-15")
spod_download_data("od", "distr", dates = dates, return_output = FALSE)
spod_convert_od_v1_to_duckdb("distr", quiet = FALSE, overwrite = TRUE)
```

**Comment by Robinlovelace** [comment]
Great to see this @e-kotov, taking a look before meeting today.

File: R/get_v1_data.R [hunk 1]
```diff
@@ -128,25 +128,28 @@ spod_available_data_v1 <- function(data_dir = spod_get_data_dir(),
 #' It can retrieve either "distritos" or "municipios" zones data.
 #'
 #' @param data_dir The directory where the data is stored.
-#' @param zones The zones for which to download the data. Can be `"districts"` (or `"dist"`, `"distr"`) or `"municipalities"` (or `"muni"`, `"municip"`).
+#' @param zones The zones for which to download the data. Can be `"districts"` (or `"dist"`, `"distr"`, or the original Spanish `"distritos"`) or `"municipalities"` (or `"muni"`, `"municip"`, or the original Spanish `"municipios"`).
```
Comment (Robinlovelace) [inline R/get_v1_data.R:131]: OK there's the explanation 👍

File: R/get_v1_data.R [hunk 2]
```diff
@@ -209,35 +212,157 @@ spod_clean_zones_v1 <- function(zones_path) {
...
+  if (is.character(dates)) {
+    if ( dates != "cached" ) {
+      con <- spod_duckdb_filter_by_dates(con, "od_csv_clean", "od_csv_clean_filtered", dates)
+    }
+  }
+  
 
-
-  # read data from cached files
+  # DBI::dbListTables(con) # for debugging only
```
Comment (Robinlovelace) [inline R/get_v1_data.R:337]: This should probably moves somewhere else, e.g. `data-raw/ad-hoc-tests`.

File: inst/extdata/sql-queries/provinces-enum.txt
Comment (Robinlovelace) [inline inst/extdata/sql-queries/provinces-enum.txt:1]: 👍

File: inst/extdata/sql-queries/when-recode-provinces.txt
Comment (Robinlovelace) [inline inst/extdata/sql-queries/when-recode-provinces.txt:1]: Is this not SQL?

File: vignettes/work-with-v1-data.qmd [hunk 3]
```diff
@@ -0,0 +1,177 @@
+---
+title: "Working with v1 MITMA data"
```
Comment (Robinlovelace) [inline vignettes/work-with-v1-data.qmd:2]: Big 👍

File: vignettes/work-with-v1-data.qmd [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
+#| include: false
+library(sf)
+library(dplyr)
+library(DBI)
+library(spanishoddata)
+```
+
+# Introduction
+
+v1 MITMA data covers the period from 2020-02-14 to 2021-05-09. Add references...
```
Comment (Robinlovelace) [inline vignettes/work-with-v1-data.qmd:43]: Worth saying what MITMA means?

File: vignettes/work-with-v1-data.qmd [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+```{r}
+DBI::dbDisconnect(od_dist_1$src$con)
+```
+
+# Get the origin-destination data for several non-consecutive dates
+
+This time let's get the data for 2020-02-14, 2020-02-17, 2021-05-07 and 2021-05-09. We will use municipalities as zones.
+
+```{r}
+od_muni_1 <- spod_get_od_v1(zones = "muni",
```
Comment (Robinlovelace) [inline vignettes/work-with-v1-data.qmd:158]: Looks like a great example.

**Review by Robinlovelace — APPROVED** [review]
Impressive amount of work in short amount of time, many thanks Egor! Minor suggestions in my review that we can think about before/after merging at your leisure.

**Comment by e-kotov** [comment]
Here's an overview of how functions are related and of the current workflow. The user only really needs (1) `spod_download_data()` and `spod_convert_od_v1_to_duckdb()` to store data in analysis ready format or (2) `spod_get_zones_v1()` if they only need a few days worth of data in a virtual duckdb table that they will aggregate immediately and load as a data frame for further analysis.

```mermaid
graph TD
    A[spod_get_data_dir] --> B[spod_get_latest_v1_file_list or spod_get_latest_v2_xml]
    B --> C[spod_available_data_v1 or spod_get_metadata]
    C --> D[spod_download_data]
    D --> E[spod_get_zones_v1 or spod_get_zones]
    
    E --> F[spod_convert_od_v1_to_duckdb]
    D --> G[spod_get_od_v1]

    F --> H[DuckDB File]
    G --> I[Virtual Table Connected to CSV Files]

    subgraph "Data Setup"
        A
        B
        C
    end

    subgraph "Data Download"
        D
        E
    end

    subgraph "Conversion to DuckDB"
        F
        H
    end

    subgraph "Virtual Table Creation"
        G
        I
    end

```

File: R/download_data.R [hunk 6]
```diff
@@ -31,9 +31,9 @@ spod_download_data <- function(
     "od", "origin-destination",
     "os", "overnight_stays",
     "tpp", "trips_per_person"),
-  zones = c("districts", "dist", "distr",
-    "municipalities", "muni", "municip",
-    "lau", "large_urban_areas"), # implement "urban_areas" for v2 data
+  zones = c("districts", "dist", "distr", "distritos",
```
[the remark this answers is not in the document]
Comment (e-kotov) [reply R/download_data.R:34]: no difference, just shortcuts. perhaps this is a bad idea)

File: R/get_v1_data.R (hunk 2 above)
Comment (e-kotov) [reply R/get_v1_data.R:337]: Will cleanup, but for debugging while stepping through the function manually this is immensely convenient. There are plenty more comments and commented code to remove.

File: inst/extdata/sql-queries/when-recode-provinces.txt
Comment (e-kotov) [reply inst/extdata/sql-queries/when-recode-provinces.txt:1]: it is, but it is just a partial query, not full query. Are you suggesting file ext = sql?

File: vignettes/work-with-v1-data.qmd (hunk 4 above)
Comment (e-kotov) [reply vignettes/work-with-v1-data.qmd:43]: Will do

File: inst/extdata/sql-queries/when-recode-provinces.txt
Comment (e-kotov) [reply inst/extdata/sql-queries/when-recode-provinces.txt:1]: filed an issue to keep that in mind #22

Comment (Robinlovelace) [reply inst/extdata/sql-queries/when-recode-provinces.txt:1]: > it is, but it is just a partial query, not full query. Are you suggesting file ext = sql?

Yes, better than .txt, even if not a fully fledged query.

**Comment by e-kotov** [comment]
Partially fulfils #5

**Pull request merged**